### PR TITLE
AIDEM: Added use to macros library

### PR DESCRIPTION
**Pull request by Wazabit**

### CODE REVIEW

File: adapters/aidem/aidem.go [hunk 1]
```diff
@@ -4,37 +4,49 @@ import (
...
 	"github.com/prebid/openrtb/v19/openrtb2"
-	"github.com/prebid/prebid-server/v2/adapters"
-	"github.com/prebid/prebid-server/v2/config"
-	"github.com/prebid/prebid-server/v2/errortypes"
-	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/v2/macros"
```
Comment (SyntaxNode) [inline adapters/aidem/aidem.go]: You must reference the /v2 packages. Based on some other changes in this PR, it appears you need to merge with the `master` branch.

File: adapters/aidem/aidem.go [hunk 2]
```diff
@@ -4,37 +4,49 @@ import (
...
 )
 
-type adapter struct {
-	endpoint string
+type AIDEMAdapter struct {
```
Comment (SyntaxNode) [inline adapters/aidem/aidem.go:17]: Please keep this as `adapter`. There is no need to export the struct by starting with a capital letter. There's no reason to repeat the adapter name as it's already in the package name.

File: adapters/aidem/aidem.go [hunk 3]
```diff
@@ -100,3 +118,25 @@ func getMediaTypeForBid(bid openrtb2.Bid) (openrtb_ext.BidType, error) {
...
 }
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpAIDEM, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	var AIDEMExt openrtb_ext.ExtImpAIDEM
```
Comment (SyntaxNode) [inline adapters/aidem/aidem.go:129]: It is customary for variables in Go to be written in lower case or camelCase.

File: static/bidder-info/aidem.yaml [hunk 4]
```diff
@@ -15,5 +15,5 @@ capabilities:
       - native
 userSync:
   redirect:
-    url: https://gum.aidemsrv.com/prebid_sync?gdpr={{.GDPR}}&consent={{.GDPRConsent}}&us_privacy={{.USPrivacy}}&redirect={{.RedirectURL}}
+    url: https://gum.aidemsrv.com/prebid_sync?gdpr={{.GDPR}}&consent={{.GDPRConsent}}&us_privacy={{.USPrivacy}}&redirect={{.RedirectURL}}&gpp={{.GPP}}&gpp_sid={{.GPPSID}}
```
Comment (gargcreation1992) [inline static/bidder-info/aidem.yaml:18]: usersync url works. 👍

**Review by gargcreation1992 — APPROVED** [review]

**Review by Sonali-More-Xandr — APPROVED** [review]

**Pull request merged**